src/data/equiv/basic.lean

```diff
@@ -1750,6 +1750,11 @@ lemma of_left_inverse_eq_of_injective {α β : Type*} [nonempty α]
   of_left_inverse f f_inv hf = of_injective f (hf ‹_›).injective :=
 by { ext, simp }
 
+lemma of_left_inverse'_eq_of_injective {α β : Type*} [nonempty α]
+  (f : α → β) (f_inv : β → α) (hf : left_inverse f_inv f) :
+  of_left_inverse' f f_inv hf = of_injective f hf.injective :=
+of_left_inverse_eq_of_injective _ _ _
+
```

Comment:
I'm pretty sure that `[nonempty α]` isn't needed to prove either this lemma or the one above